docs/docs/store.mdx

```diff
@@ -37,48 +37,134 @@ When you want to update the store, you can call the `update()` method passing th
 
 ```ts title="session.service.ts"
 import { SessionStore } from './session.store';
- 
+
 export class SessionService {
   constructor(private sessionStore: SessionStore) {}
 
   updateUserName(newName: string) {
     this.sessionStore.update({ name: newName });
-  }  
+  }
 }
 ```
 
 The second `update()` option gives you more control. It receives a `callback` function, which gets the current state, and returns a new **immutable** state, which will be the new value of the store. For example:
 
 ```ts title="session.service.ts"
 import { SessionStore } from './session.store';
- 
+
 export class SessionService {
   constructor(private sessionStore: SessionStore) {}
 
   updateUserName(newName: string) {
     this.sessionStore.update(state => ({
       name: newName
     }));
-  }  
+  }
+}
+```
+
+### `overwrite()`
+
+When you don't just want to update the current store state but replace the state completely, you can call the `overwrite()` method passing the new `state`:
+
+```ts title="session.service.ts"
+import { SessionStore } from './session.store';
+
+export class SessionService {
+  constructor(private sessionStore: SessionStore) {}
+
+  updateUserName(newName: string) {
+    this.sessionStore.overwrite({ name: newName });
+  }
+}
+```
+
+As with `update()` the `overwrite()` method also has a second option which gives you more control. It receives a `callback` function, which gets the current state, and returns a new **immutable** state, which will be the new value of the store. For example:
+
+```ts title="session.service.ts"
+import { SessionStore } from './session.store';
+
+export class SessionService {
+  constructor(private sessionStore: SessionStore) {}
+
+  updateUserName(newName: string) {
+    this.sessionStore.overwrite(state => ({
+      name: newName
+    }));
+  }
 }
 ```
 
+You should use `overwrite()` over `update()` when you want to completely replace the current state at the top level.
+
+```ts title="session.store.ts"
+type SessionState = SessionStateAuthenticated | SessionStateUnauthenticated;
+
+interface SessionStateAuthenticated {
+  _tag: 'session-state-authenticated';
```

Comment:
I prefer using a simple example state and not copy-paste from your application.

Reply:
This was a much reduced version of the original, I though that session state was a good simple example and "real world" enough to be useful. My preference when reading documentation has always been simplified "real world" over slightly contrived.

I could simplify the example further to

```
type SessionState = AuthenticatedState | UnauthenticatedState;

interface AuthenticatedState {
  _tag: 'authenticated';
  uid: string;
}

interface UnauthenticatedState {
  _tag: 'unauthenticated'
}
```

or even do something like

```
type SessionState = TokenState | InitialState;

interface TokenState {
  name: string;
  token: string;
}

interface InitialState{
  name: string;
}
```

Reply:
I prefer the latter

Reply:
I'll change to the second option.